### PR TITLE
SAK-40004 - Replace org.hibernate.dialect.function

### PR DESCRIPTION
**Pull request by deinok**
Replace Deprecated functions of Hibernate 4.X in favour of the supported functions

### CODE REVIEW

File: kernel/kernel-private/src/main/java/org/sakaiproject/springframework/orm/hibernate/AddableSessionFactoryBean.java [hunk 1]
```diff
@@ -27,9 +27,7 @@
 
 import org.hibernate.SessionFactory;
 import org.hibernate.cfg.Configuration;
-import org.hibernate.dialect.function.ClassicAvgFunction;
-import org.hibernate.dialect.function.ClassicCountFunction;
-import org.hibernate.dialect.function.ClassicSumFunction;
+import org.hibernate.dialect.function.*;
```
[the remark this answers is not in the document]
Comment (bjones86) [reply kernel/kernel-private/src/main/java/org/sakaiproject/springframework/orm/hibernate/AddableSessionFactoryBean.java]: Please don't use `*` imports

File: kernel/kernel-private/src/main/java/org/sakaiproject/springframework/orm/hibernate/AddableSessionFactoryBean.java [hunk 2]
```diff
@@ -46,8 +44,8 @@
  *
  */
 @Slf4j
-public class AddableSessionFactoryBean extends LocalSessionFactoryBean implements ApplicationContextAware
-{
+public class AddableSessionFactoryBean extends LocalSessionFactoryBean implements ApplicationContextAware {
+
```
Comment (bjones86) [inline kernel/kernel-private/src/main/java/org/sakaiproject/springframework/orm/hibernate/AddableSessionFactoryBean.java]: I think this change is unncessary

File: kernel/kernel-private/src/main/java/org/sakaiproject/springframework/orm/hibernate/AddableSessionFactoryBean.java [hunk 3]
```diff
@@ -87,4 +85,5 @@ protected SessionFactory buildSessionFactory(LocalSessionFactoryBuilder sfb) {
 
 		return sfb.buildSessionFactory();
 	}
+
```
Comment (bjones86) [inline kernel/kernel-private/src/main/java/org/sakaiproject/springframework/orm/hibernate/AddableSessionFactoryBean.java]: Same here

File: kernel/kernel-private/src/main/java/org/sakaiproject/springframework/orm/hibernate/AddableSessionFactoryBean.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonespm) [reply kernel/kernel-private/src/main/java/org/sakaiproject/springframework/orm/hibernate/AddableSessionFactoryBean.java]: I don't even think this is needed if you're removing all the functions either

**Comment by deinok** [comment]
@bjones86 Sorry, IDE do this kind of things for me sometimes. Changes requested done

**Comment by bjones86** [comment]
Thanks @deinok!

**Pull request merged**